### ISSUE TITLE
Custom start and end date for Segment and Subsegment

### ISSUE THREAD

**Issue by okubiak** [aws/aws-xray-sdk-dotnet#76]
Hi,

I'm currently facing a little issue trying to profile Elasticache Redis commands.
I'm using StackExchange.Redis client and the profiling system included is not really friendly with the AWSXRayRecorder design.

All what I can have is a list of commands elapsed time.
I tried to set the start and end times for a subsegment but `AWSXRayRecorder.Instance.EndSubsegment();` is overriding the end time with utcnow (which is fair).
For now I just added the elapsed time in the subsegment metadata but it will be complicate to investigate later in XRay console.
```
recorder.BeginSubsegment("Redis Commands");
recorder.SetNamespace("Database");
recorder.AddMetadata(nameof(command.Command), command.Command);
recorder.AddMetadata(nameof(command.ElapsedTime), command.ElapsedTime);
AWSXRayRecorder.Instance.EndSubsegment();
```

So my first question is: Do you think that customizing start and end time would be a solution at some point?
As StackExchange.Redis does not provide any callback where I could call a `BeginSubsegment` and `EndSubsegment`...

Second question: Do you have any other recommendation to profile Redis?

Regards

**Comment by yogiraj07** [comment]
Hi @okubiak ,
1. Currently [BeginSegment()](https://github.com/aws/aws-xray-sdk-dotnet/blob/master/sdk/src/Core/AWSXRayRecorderImpl.cs#L146) and [EndSegment()](https://github.com/aws/aws-xray-sdk-dotnet/blob/master/sdk/src/Core/AWSXRayRecorderImpl.cs#L188) accepts custom start and end time for the `Segment`. For your use case, we need to add similar support for `BeginSubsegment()` and `EndSubsegment()` which will ease the user experience.

2. For second question, `StackExchange.Redis` is a custom client and I think surrounding the call with custom `Subsegment` is the way to go. 

I will keep this issue open and tag it as a feature request. You are welcome to submit a PR.

Thanks,
Yogi

**Comment by srprash** [comment]
Hi @okubiak ,
This feature has been implemented in v2.7.0
Please refer to the [Changelog](https://github.com/aws/aws-xray-sdk-dotnet/blob/master/CHANGELOG.md) for details.

Thanks,
Prashant
